api/v1alpha1/project_types.go

```diff
@@ -50,7 +50,25 @@ func (p *Project) GetStatus() *ProjectStatus {
 
 // ProjectSpec describes a Project.
 type ProjectSpec struct {
-	// TODO: Figure out the attributes of a ProjectSpec.
+	// PromotionPolicies defines policies governing the promotion of Freight to
+	// specific Stages within this Project.
+	PromotionPolicies []PromotionPolicy `json:"promotionPolicies,omitempty"`
+}
+
+// PromotionPolicy defines policies governing the promotion of Freight to a
+// specific Stage.
+type PromotionPolicy struct {
+	//+kubebuilder:validation:MinLength=1
+	//+kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$
+	Stage string `json:"stage"`
+	// EnableAutoPromotion indicates whether new Freight can automatically be
+	// promoted into the Stage referenced by the Stage field. Note: There are may
+	// be other conditions also required for an auto-promotion to occur. This
+	// field defaults to false, but is commonly set to true for Stages that
+	// subscribe to Warehouses instead of other, upstream Stages. This allows
+	// users to define Stages that are automatically updated as soon as new
+	// artifacts are detected.
+	EnableAutoPromotion bool `json:"enableAutoPromotion,omitempty"`
```

Comment:
Can we use this as an opportunity to rename this to:

```
  autoPromotionEnabled: true
```

I think `Enabled` as a suffix is more canonical kubernetes. e.g.:

```
  sslEnabled
  dhcpV6Enabled
```

Reply:
Consider it done.

Reply:
Addressed in the last six commits.